plugins/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorTests.java

```diff
@@ -237,9 +241,141 @@ public void testNonExistentWithoutIgnoreMissing() throws Exception {
+        assertThat(rawMetadata.get("access_permission:modify_annotations"), is("true"));
+        assertThat(rawMetadata.get("access_permission:can_print_degraded"), is("true"));
+        assertThat(rawMetadata.get("meta:creation-date"), is("2016-09-30T13:19:58Z"));
+        // "created" is different depending on the JVM Locale. We just test that it's not null
+        assertThat(rawMetadata.get("created"), is(notNullValue()));
+        assertThat(rawMetadata.get("access_permission:extract_for_accessibility"), is("true"));
+        assertThat(rawMetadata.get("access_permission:assemble_document"), is("true"));
+        assertThat(rawMetadata.get("xmpTPg:NPages"), is("1"));
+        assertThat(rawMetadata.get("Creation-Date"), is("2016-09-30T13:19:58Z"));
+        assertThat(rawMetadata.get("dcterms:created"), is("2016-09-30T13:19:58Z"));
+        assertThat(rawMetadata.get("dc:format"), is("application/pdf; version=1.4"));
+        assertThat(rawMetadata.get("access_permission:extract_content"), is("true"));
+        assertThat(rawMetadata.get("access_permission:can_print"), is("true"));
+        assertThat(rawMetadata.get("pdf:docinfo:creator_tool"), is("Writer"));
+        assertThat(rawMetadata.get("access_permission:fill_in_form"), is("true"));
+        assertThat(rawMetadata.get("pdf:encrypted"), is("false"));
+        assertThat(rawMetadata.get("producer"), is("LibreOffice 5.2"));
+        assertThat(rawMetadata.get("access_permission:can_modify"), is("true"));
+        assertThat(rawMetadata.get("pdf:docinfo:producer"), is("LibreOffice 5.2"));
+        assertThat(rawMetadata.get("pdf:docinfo:created"), is("2016-09-30T13:19:58Z"));
+        assertThat(rawMetadata.get("Content-Type"), is("application/pdf"));
+    }
+
+    public void testRawMetadataFromRtf() throws Exception {
+        processor = new AttachmentProcessor(randomAsciiOfLength(10), "source_field",
+            "target_field", EnumSet.of(AttachmentProcessor.Property.RAW_METADATA), 10000, false);
+        Map<String, Object> attachmentData =
+            parseBase64Document("e1xydGYxXGFuc2kNCkxvcmVtIGlwc3VtIGRvbG9yIHNpdCBhbWV0DQpccGFyIH0=", processor);
+        assertThat(attachmentData.keySet(), contains("raw_metadata"));
+        assertThat(attachmentData.get("raw_metadata"), is(instanceOf(Map.class)));
+        @SuppressWarnings("unchecked")
+        Map<String, Object> rawMetadata = (Map<String, Object>) attachmentData.get("raw_metadata");
+
+        assertThat(rawMetadata.keySet(), containsInAnyOrder("X-Parsed-By", "Content-Type"));
+
+        assertThat(rawMetadata.get("X-Parsed-By"), is("org.apache.tika.parser.rtf.RTFParser"));
+        assertThat(rawMetadata.get("Content-Type"), is("application/rtf"));
+    }
+
     private Map<String, Object> parseDocument(String file, AttachmentProcessor processor) throws Exception {
+        return parseBase64Document(getAsBase64(file), processor);
+    }
+
+    // Adding this method to more easily write the asciidoc documentation
```

Comment:
I just let it here as a comment. It helps developers to easily add a new type of file in the test suite, collects all assertions in the logs, then copy and paste the log in the test case itself.